fix: Skip specific errors when parsing yaml IaC files

ipapast opened the pull request:
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules

#### What does this PR do?

We identified a few cases that our services parse the yaml files in a different way. The node 'yaml' library is too 'strict', returning errors for validation of the YAML as well. 

In this PR we focus on two specific errors: bad indentation and duplicate keys. 
These two examples are not consistent with what Policy Engine returns so in this commit we intentionally skip errors for these cases, and return the parsed YAML document instead.

This will improve the consistency between the two parsers.

#### Where should the reviewer start?
The changes are in `file-parser.ts` and `iac-parser.ts` where we check if the error is one of the two error strings.

#### How should this be manually tested?
- Create a file with a SemanticError, e.g. duplicate keys:
```
apiVersion: v1
kind: Pod
metadata:
  name: myapp-pod
spec:
  containers:
  something: here
metadata:
  another: thing
```

Run `snyk-dev iac test duplicate_keys.yaml`
See that it is no longer failing but showing results (if any).

#### What are the relevant tickets?
[CC-927]

#### Screenshots
Before:
![image](https://user-images.githubusercontent.com/6989529/120799626-e96be400-c536-11eb-854c-2deadb406caf.png)

After:
![image](https://user-images.githubusercontent.com/6989529/120799647-eec92e80-c536-11eb-980d-74507269a59f.png)


#### Additional questions


[CC-927]: https://snyksec.atlassian.net/browse/CC-927

ipapast commented on src/cli/commands/test/iac-local-execution/file-parser.ts:
```diff
@@ -69,7 +69,11 @@ function parseYAMLOrJSONFileData(fileData: IacFileData): any[] {
+      // we decided to fail only on YAMLSyntaxError to be consistent with the Golang parser in Policy Engine.
+      if (doc.errors.length !== 0
+        && doc.errors[0].name === 'YAMLSyntaxError'
+      || !doc.errors[0].message.includes('expected flow map')) {
```
I don't feel great about this line. I added it temporarily to discuss on Monday.
In this case: `{ "foo": "bar"` this would produce a FlowMap error, which is not a SyntaxError, but a SemanticError.

We need to think again how we approach these and compare them with Policy Engine.

almog27 commented on src/cli/commands/test/iac-local-execution/file-parser.ts:
```diff
@@ -70,7 +70,15 @@ function parseYAMLOrJSONFileData(fileData: IacFileData): any[] {
-        throw doc.errors[0];
+        // the YAML Parser is more strict than the Golang one in Policy Engine,
+        // so we decided to skip specific errors in order to be consistent
+        const errorsToSkip = [
+          'Insufficient indentation in flow collection',
+          'Map keys must be unique',
+        ];
```
Can we extract this const outside of the map, maybe to a more shared place so it could be also used [here](https://github.com/snyk/snyk/pull/2003/files#diff-4e236e93db42d2bea509d66fbf89ab3bb0ac064b9ea08df6cf3e97a017f5ac5fR37-R40)

teodora-sandu commented on test/jest/unit/iac-unit-tests/file-parser.spec.ts:
```diff
@@ -119,12 +121,19 @@ describe('parseFiles', () => {
 
+  // the npm yaml parser by default fails on SemanticErrors like duplicate keys
+  // but we decided to skip this error in order to be consistent with the Policy Engine
+  it('parses files with SemanticYAMLError because of duplicate key correctly', async () => {
```
Can we also add a test case for  the`'Insufficient indentation in flow collection'` error? So that if the library somehow changes the error message, we are notified about it and know to update it

ipapast commented:
@almog27 @teodora-sandu  I extracted the check as a function and gave it a descriptive name, also added an extra test. If you could have a look again 🙇
